src/main/java/de/zalando/aruha/nakadi/controller/EventTypeController.java

```diff
@@ -65,6 +66,12 @@ public EventTypeController(EventTypeRepository eventTypeRepository, TopicReposit
         }
 
         try {
+            // can happen if EventType already exists or if topic deletion
+            // was not yet processed after EventType deletion
+            if (topicRepository.topicExists(eventType.getName())) {
```

Comment:
Another approach would be to change the code for `topicRepository.createTopic` as follows:

```java
@Override
    public void createTopic(final String topic, final int partitionsNum, final int replicaFactor,
                            final long retentionMs, final long rotationMs) throws TopicCreationException {
        ZkUtils zkUtils = null;
        try {
            final String connectionString = zkFactory.get().getZookeeperClient().getCurrentConnectionString();
            zkUtils = ZkUtils.apply(connectionString, settings.getZkSessionTimeoutMs(),
                    settings.getZkConnectionTimeoutMs(), false);

            final Properties topicConfig = new Properties();
            topicConfig.setProperty("retention.ms", Long.toString(retentionMs));
            topicConfig.setProperty("segment.ms", Long.toString(rotationMs));

            AdminUtils.createTopic(zkUtils, topic, partitionsNum, replicaFactor, topicConfig);
        } catch (TopicExistsException e) {
            throw new DuplicatedEventTypeNameException("EventType with name " + topic +
                        " already exists (or wasn't completely removed yet)");
        } catch (Exception e) {
            throw new TopicCreationException("Unable to create topic", e);
        }
        finally {
            if (zkUtils != null) {
                zkUtils.close();
            }
        }
    }
```


Then you would not be required to change the controller code. This can be handy if we are using the repository outside of the controller context.